includes/Taxonomy/Tag_Taxonomy.php

```diff
@@ -0,0 +1,98 @@
+
+use Google\Web_Stories\REST_API\Stories_Terms_Controller;
+use Google\Web_Stories\Story_Post_Type;
+
+/**
+ * Tag_Taxonomy class.
+ */
+class Tag_Taxonomy extends Taxonomy_Base {
+	/**
+	 * @var string
+	 */
+	protected $taxonomy_slug = 'web-story-tag';
+	/**
+	 * @var string
+	 */
+	protected $taxonomy_post_type = Story_Post_Type::POST_TYPE_SLUG;
+	/**
+	 * Taxonomy args.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return array
+	 */
+	protected function taxonomy_args() : array {
+		$labels = [
+			'name'                       => _x( 'Tags', 'taxonomy general name', 'web-stories' ),
+			'singular_name'              => _x( 'Tag', 'taxonomy singular name', 'web-stories' ),
+			'search_items'               => __( 'Search Tags', 'web-stories' ),
+			'popular_items'              => __( 'Popular Tags', 'web-stories' ),
+			'all_items'                  => __( 'All Tags', 'web-stories' ),
+			'parent_item'                => __( 'Parent Tag', 'web-stories' ),
+			'parent_item_colon'          => __( 'Parent Tag:', 'web-stories' ),
+			'edit_item'                  => __( 'Edit Tag', 'web-stories' ),
+			'view_item'                  => __( 'View Tag', 'web-stories' ),
+			'update_item'                => __( 'Update Tag', 'web-stories' ),
+			'add_new_item'               => __( 'Add New Tag', 'web-stories' ),
+			'new_item_name'              => __( 'New Tag Name', 'web-stories' ),
+			'separate_items_with_commas' => __( 'Separate Tags with commas', 'web-stories' ),
+			'add_or_remove_items'        => __( 'Add or remove Tags', 'web-stories' ),
+			'choose_from_most_used'      => __( 'Choose from the most used Tags', 'web-stories' ),
+			'not_found'                  => __( 'No Tags found.', 'web-stories' ),
+			'no_terms'                   => __( 'No Tags', 'web-stories' ),
+			'filter_by_item'             => __( 'Filter by Tag', 'web-stories' ),
+			'items_list_navigation'      => __( 'Tags list navigation', 'web-stories' ),
+			'items_list'                 => __( 'Tags list', 'web-stories' ),
+			/* translators: Tab heading when selecting from the most used terms. */
+			'most_used'                  => _x( 'Most Used', 'Tags', 'web-stories' ),
+			'back_to_items'              => __( '&larr; Go to Tags', 'web-stories' ),
+			'item_link'                  => _x( 'Tag Link', 'navigation link block title', 'web-stories' ),
+			'item_link_description'      => _x( 'A link to a Tag.', 'navigation link block description', 'web-stories' ),
+		];
+		$args   = [
+			'labels'                => $labels,
+			'hierarchical'          => false,
+			'public'                => false,
+			'show_ui'               => true,
+			'show_admin_column'     => true,
+			'rewrite'               => true,
+			'show_in_rest'          => true,
+			'capabilities'          => [
+				'manage_terms' => 'delete_private_web-stories',
+				'edit_terms'   => 'delete_private_web-stories',
+				'delete_terms' => 'delete_private_web-stories',
+				'assign_terms' => 'edit_web-stories',
+			],
```

Comment:
I mapped this functionality to existing capabilities. Do we need to custom capabilities for the taxonomies as well?

Reply:
Good question.

Curious: why the `delete_private_web-stories` choice? Just soo it's limited to admins/editors?
Aside: Shouldn't these come from the post type object?

Since we already have a `Capabilities` class and DB migration functionality, adding custom capabilities seems straightforward. So `manage_web-story_categories`, `edit_web-story_categories`, and `deelte_web-story_tags`, ..., etc. WDYT? Would that work?

Orr perhaps the same set of capabilities for both cats and tags to keep things simple, e.g. `manage_web-story_terms`, `delete_web-story_terms`, etc.

Reply:
Have added some basic default capability. It did make a lot of sense to have fine gain capabilities for different taxnomies. I have updated the PR with these new capabilities.